packages/office-addin-test-server/src/testServer.ts

```diff
@@ -24,46 +24,58 @@ export class TestServer {
         this.m_resultsPromise = undefined;
         this.m_testServerStarted = false;
     }
-    
-    public async startTestServer(mochaTest: boolean = false): Promise<any>{
-        return new Promise<boolean>(async (resolve, reject) => {
+
+    public async startTestServer(mochaTest: boolean = false): Promise<any> {
+        try {
             if (mochaTest) {
                 process.env["NODE_TLS_REJECT_UNAUTHORIZED"] = "0";
             }
 
+            // create express server instance
             const options = await devCerts.getHttpsServerOptions();
-            const platformName = this.getPlatformName();
-
-            // listen for 'ping'
             this.m_app.use(cors());
-            this.m_app.get("/ping", function (req: any, res: any, next: any) {                
-                res.send(platformName);
-            });
-
-            // listen for posting of test results
-            this.m_resultsPromise = new Promise<JSON>(async (resolveResults) => {
-                this.m_app.post("/results", async (req: any, res: any) => {
-                    res.send("200");
-                    this.m_jsonData = JSON.parse(req.query.data);
-                    resolveResults(this.m_jsonData);
-                });
-            });
-    
             this.m_server = https.createServer(options, this.m_app);
-    
+
             // start listening on specified port
+            return await this.startListening();
+
+        } catch (err) {
+            throw new Error(`Test server failed with error.\n${err}`);
+        }
+    }
+
+    private async startListening(): Promise<any> {
+        return new Promise<any>(async (resolve, reject) => {
             try {
+                // set server to listen on specified port
                 this.m_server.listen(this.m_port, () => {
                     this.m_testServerStarted = true;
                     resolve(true);
                 });
+
+                // listen for 'ping'
+                const platformName = this.getPlatformName();
+                this.m_app.get("/ping", function (req: any, res: any, next: any) {
+                    res.send(platformName);
+                    resolve(true);
+                });
+
+                // listen for posting of test results
+                this.m_resultsPromise = new Promise<JSON>(async (resolveResults) => {
```

Comment:
I don't think this code needs to go inside here.

Reply:
but I think it can safely there and really it does fall in the category of a listening event